libs/SmartStore/SmartStoreTests/SFSmartSqlTests.m

```diff
@@ -167,6 +173,27 @@ - (void) testConvertSmartSqlWithSpecialColumnsAndSelfJoin
                          [self.store convertSmartSql:@"select mgr.{employees:_soupEntryId}, e.{employees:_soupEntryId} from {employees} as mgr, {employees} as e"], @"Bad conversion");
 }
 
+- (void) testConvertSmartSqlWithJSON1
+{
+    XCTAssertEqualObjects(@"select TABLE_1_1, json_extract(soup, '$.education') from TABLE_1 where json_extract(soup, '$.education') = 'MIT'",
+                          [self.store convertSmartSql:@"select {employees:lastName}, {employees:education} from {employees} where {employees:education} = 'MIT'"], @"Bad conversion");
+}
+
+- (void) testConvertSmartSqlWithJSON1AndTableQualifiedColumn
+{
+    XCTAssertEqualObjects(@"select json_extract(TABLE_1.soup, '$.education') from TABLE_1 order by json_extract(TABLE_1.soup, '$.education')",
+                          [self.store convertSmartSql:@"select {employees}.{employees:education} from {employees} order by {employees}.{employees:education}"], @"Bad conversion");
+}
+
+- (void) testConvertSmartSqlWithJSON1AndTableAliases
+{
+    XCTAssertEqualObjects(@"select json_extract(e.soup, '$.education'), json_extract(soup, '$.building') from TABLE_1 as e, TABLE_2",
+                          [self.store convertSmartSql:@"select e.{employees:education}, {departments:building} from {employees} as e, {departments}"], @"Bad conversion");
+    
+    // XXX join query with json1 will only run if all the json1 columns are qualified by table or alias
```

Comment:
If one writes a smart sql joining on json1 fields, because they do json_extract against the soup column which exist in all soup tables, it will not run. The work-around is for the user to do {soup}.{soup:json1path} or alias.{soup:json1path} (assuming the soup table was aliased in the FROM clause). We should look into making the smart sql to sql generation smarter to do json_extract(table.soup, ). It might not be doable without fully parsing the query though.